File: LeetCode/Easy/1512-number-of-good-pairs/1512-number-of-good-pairs.go
func numIdenticalPairs(nums []int) int {
    count := make([]int, 101)
    ret := 0

    for _, v := range nums { count[v]++ }

    for i := 1; i < 101; i++ {
        ret += count[i] * (count[i] - 1) / 2
    }

    return ret
}

File: LeetCode/Easy/1512-number-of-good-pairs/1512-number-of-good-pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        vector<int> count(101);
        int ret = 0;
        
        for (int v : nums) ++count[v];
        
        for (int i = 1; i < 101; ++i)
            ret += count[i] * (count[i] - 1) / 2;
        
        return ret;
    }
};